### PR TITLE
Allow adding collaborators with (fullname)

### PR DESCRIPTION
**Pull request by svarlamov**
Signed-off-by: Sasha Varlamov <sasha@sashavarlamov.com>

Targeting: #3102

Perhaps we should fork this out into a utility func? I'm pretty new to Gitea, so I just in-lined it same as #2973 but since this keeps popping up, I suppose we might want a shared function for this... Where should I put it and then refactor the #2973 fix?

### CODE REVIEW

**Comment by lunny** [comment]
LGTM

File: routers/repo/setting.go [hunk 1]
```diff
@@ -367,6 +367,10 @@ func Collaboration(ctx *context.Context) {
 // CollaborationPost response for actions for a collaboration of a repository
 func CollaborationPost(ctx *context.Context) {
 	name := strings.ToLower(ctx.Query("collaborator"))
+	// name may be formatted as "username (fullname)"
+	if strings.Contains(name, "(") && strings.HasSuffix(name, ")") {
```
Comment (ethantkoenig) [inline routers/repo/setting.go]: nit: can simplify to
```
if index := strings.Index(name, " ("); index >= 0 {
	name = name[:index]
}

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: routers/repo/setting.go (hunk 1 above)
Comment (svarlamov) [reply routers/repo/setting.go]: I just cut-n-pasted @lunny's fix. I agree with you on this. Where can I put a utility method and then refactor all of the usages? Sorry new to Gitea, so want to make sure I'm following your guys' best practices... Thanks

Comment (ethantkoenig) [reply routers/repo/setting.go]: I think adding a `routers/utils/utils.go` file and putting it there makes the most sense. This proposed function seems specific to route handlers, so I'd prefer to not add it to `modules/util`.

**Review by ethantkoenig — APPROVED** [review]

**Comment by svarlamov** [comment]
So I moved it out into a new utils pkg within routers. I also refactored the two usages within `routers/org/teams.go` and `routers/repo/setting.go`. I'm sure that there are probably more places that need to have this added, but maybe that's a separate PR. Let's get this merged because it's a pretty big issue for users...

**Comment by ethantkoenig** [comment]
LGTM. Thank you for adding tests!

**Comment by lunny** [comment]
make L-G-T-M work

**Comment by lunny** [comment]
Please send a back port to branch `release/v1.3`

**Comment by svarlamov** [comment]
@lunny So I should file another PR against that rel branch? New to the Gitea standards here :)

**Comment by ethantkoenig** [comment]
@svarlamov Yes, see https://github.com/go-gitea/gitea/pull/3106 as an example of a backport.

**Comment by lunny** [comment]
@svarlamov
I just finished the back port locally:
```Shell
git fetch release/v1.3
git co release/v1.3
git co -b lunny/fix_xxx
git cherry-pick xxxx
......
```

**Pull request merged**